### PR TITLE
fix: deterministic results for address.nearbyGPSCoordinate

### LINKED ISSUE #736: `address.nearbyGPSCoordinate` non deterministic results
### Describe the bug

`address.nearbyGPSCoordinate` generates non-deterministic values when reference coordinates are given.

### Reproduction

```
for (let i = 0; i < 5; i++) {
  faker.seed(522003803);

  const coordinate = faker.address.nearbyGPSCoordinate([0, 0]);

  console.log(coordinate);
}
```

You will see 5 different string tuples being printed to the console.

Stackblitz URL:
https://stackblitz.com/edit/typescript-hm6h4k

### Additional Info

`address.nearbyGPSCoordinate` uses `Math.random` when generating the bearing for random coordinates.
https://github.com/faker-js/faker/blob/7afd8c55baac6f7d895caabeb623cc6a0514d6cb/src/address.ts#L539-L544

Also, this can cause the tests randomly to fail. 
That's how I noticed this.

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
closes #736

### CODE REVIEW

**Review by pkuczynski — DISMISSED** [review]

File: test/address.spec.ts [hunk 1]
```diff
@@ -333,6 +351,17 @@ describe('address', () => {
           expect(timeZone).toEqual(expectations.timeZone);
         });
       });
+
+      describe('nearbyGPSCoordinate()', () => {
+        it('returns expected coordinates', () => {
+          faker.seed(seed);
+
+          const coords = faker.address.nearbyGPSCoordinate(
+            inputs.nearbyGpsCoordinates
```
Comment (ST-DDT) [inline test/address.spec.ts:360]: Maybe just hardcode them here? The value is always the same AFAICT.

File: src/address.ts [hunk 2]
```diff
@@ -538,7 +538,13 @@ export class Address {
     // This approach will likely result in a higher density of points near the center.
     const randomCoord = coordinateWithOffset(
       coordinate,
-      degreesToRadians(Math.random() * 360.0),
+      degreesToRadians(
+        this.faker.datatype.number({
+          min: 0,
+          max: 1,
+          precision: 1e-4,
+        }) * 360.0
```
[the remark this answers is not in the document]
Comment (Shinigami92) [reply src/address.ts:546]: I would just go with e.g. `faker.datatype.float({ min: 0, max: 360, precision: 1e-4 })`

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: src/address.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (xDivisionByZerox) [reply src/address.ts]: Updated in f81341e83c2f3f0267d882f7c05550f1907a17b2

File: test/address.spec.ts (hunk 1 above)
Comment (xDivisionByZerox) [reply test/address.spec.ts]: Updated in c0258cb4af0eaced6606689d4c4d6c8caf83d8a4

File: test/address.spec.ts [hunk 3]
```diff
@@ -29,6 +29,12 @@ const seededRuns = [
       cardinalDirection: 'East',
       cardinalDirectionAbbr: 'E',
       timeZone: 'Europe/Amsterdam',
+      nearbyGpsCoordinates: ['-0.0393', '0.0396'],
+    },
+    inputs: {
+      nearbyGpsCoordinates: [0, 0],
+    } as {
+      nearbyGpsCoordinates: [number, number];
```
[the remark this answers is not in the document]
Comment (xDivisionByZerox) [reply test/address.spec.ts]: Removed in c0258cb4af0eaced6606689d4c4d6c8caf83d8a4

**Comment by xDivisionByZerox** [comment]
I'm testing local again to see if I forgot something, but I normally always run tests locally before pushing :/

**Comment by xDivisionByZerox** [comment]
The `max = 360` implemented a rounding change on the 4th decimal place of 1 expect value :)

**Comment by xDivisionByZerox** [comment]
> The `max: 360` property in datatype.number implemented a rounding change on the 4th decimal place of 1 expect value :)

Well, that's funny;
If I set up a test by myself I get `[ '-0.0394', '0.0396' ]` as result.
```
import { faker } from './src';

faker.seed(42);
console.log(faker.address.nearbyGPSCoordinate([0, 0]));
```

But if I run the test suite it gets `[ '-0.0393', '0.0396' ]`

**Comment by Shinigami92** [comment]
Should we use 1e-5 ?

**Comment by xDivisionByZerox** [comment]
🎉🎉🎉

**Review by ST-DDT — APPROVED** [review]

**Review by import-brain — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**